controllers/nodemaintenance_controller_test.go

```diff
@@ -11,334 +12,297 @@ import (
-		req = reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Name: nm.ObjectMeta.Name,
-			},
+			Expect(k8sClient.Create(context.Background(), testNs)).To(Succeed())
 		}
-	})
-
-	AfterEach(func() {
-		stopTestEnv()
-	})
-
-	Context("Initialization test", func() {
-
-		It("Node maintenance should be initialized properly", func() {
-			_ = r.initMaintenanceStatus(nm)
-			maintenance := &nodemaintenanceapi.NodeMaintenance{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nm), maintenance)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(maintenance.Status.Phase).To(Equal(nodemaintenanceapi.MaintenanceRunning))
-			Expect(len(maintenance.Status.PendingPods)).To(Equal(2))
-			Expect(maintenance.Status.EvictionPods).To(Equal(2))
-			Expect(maintenance.Status.TotalPods).To(Equal(2))
-			Expect(maintenance.Status.DrainProgress).To(Equal(0))
-			Expect(maintenance.Status.LastUpdate.IsZero()).To(BeFalse())
-		})
-
-		It("owner ref should be set properly", func() {
-			_ = r.initMaintenanceStatus(nm)
-			maintenance := &nodemaintenanceapi.NodeMaintenance{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nm), maintenance)
-			node := &corev1.Node{}
-			err = k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, node)
-			Expect(err).ToNot(HaveOccurred())
-			r.setOwnerRefToNode(maintenance, node)
-			Expect(len(maintenance.ObjectMeta.GetOwnerReferences())).To(Equal(1))
-			ref := maintenance.ObjectMeta.GetOwnerReferences()[0]
-			Expect(ref.Name).To(Equal(node.ObjectMeta.Name))
-			Expect(ref.UID).To(Equal(node.ObjectMeta.UID))
-			Expect(ref.APIVersion).To(Equal(node.TypeMeta.APIVersion))
-			Expect(ref.Kind).To(Equal(node.TypeMeta.Kind))
-			r.setOwnerRefToNode(maintenance, node)
-			Expect(len(maintenance.ObjectMeta.GetOwnerReferences())).To(Equal(1))
-		})
-
-		It("Should not init Node maintenance if already set", func() {
-			nmCopy := nm.DeepCopy()
-			nmCopy.Status.Phase = nodemaintenanceapi.MaintenanceRunning
-			_ = r.initMaintenanceStatus(nmCopy)
-			maintenance := &nodemaintenanceapi.NodeMaintenance{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nm), maintenance)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(maintenance.Status.Phase).NotTo(Equal(nodemaintenanceapi.MaintenanceRunning))
-			Expect(len(maintenance.Status.PendingPods)).NotTo(Equal(2))
-			Expect(maintenance.Status.EvictionPods).NotTo(Equal(2))
-			Expect(maintenance.Status.TotalPods).NotTo(Equal(2))
-			Expect(maintenance.Status.DrainProgress).To(Equal(0))
-			Expect(maintenance.Status.LastUpdate.IsZero()).To(BeTrue())
-		})
-
+		nodeOne = getNode(taintedNodeName)
 	})
 
 	Context("Taint functioninality test", func() {
+		// var nm *nodemaintenanceapi.NodeMaintenance
+		BeforeEach(func() {
+			Expect(k8sClient.Create(context.Background(), nodeOne)).To(Succeed())
+			DeferCleanup(k8sClient.Delete, context.Background(), nodeOne)
+		})
```

Comment:
Adding a `When` container node would improve the readability of the test: when we expect the outcome expressed in the `It` container node